projects/step-frontend/src/lib/modules/resources/components/resource-configuration-dialog/resource-configuration-dialog.component.ts

```diff
@@ -0,0 +1,149 @@
+
+  protected uploading = false;
+  protected contentUpdated = false;
+
+  ngOnInit(): void {
+    // Disable automatic closing, switch to manual
+    this._matDialogRef.disableClose = true;
+
+    this.setFormValue(this._resourceConfigurationDialogData.resource);
+    this.initBackdropClosing();
+  }
+
+  ngOnDestroy(): void {
+    this.terminator$.next();
+    this.terminator$.complete();
+  }
+
+  protected onContentChange(resourceId: string): void {
+    if (!resourceId.startsWith('resource:')) {
+      return;
+    }
+
+    this._resourcesService.getResource(resourceId.replace('resource:', '')).subscribe((resource) => {
+      this._resourceConfigurationDialogData.resource = resource;
+      this.setFormValue(resource);
+    });
+  }
+
+  protected onFilesChange(): void {
+    this.uploading = true;
+    this.contentUpdated = true;
+  }
+
+  protected onUploadComplete(): void {
+    this.uploading = false;
+  }
+
+  protected close(): void {
+    if (!this.contentUpdated) {
+      this._matDialogRef.close();
+      return;
+    }
+
+    const resource = {
+      ...this._resourceConfigurationDialogData.resource,
+    };
+
+    resourceConfigurationDialogFormSetValueToModel(this.formGroup, resource);
+
+    this._resourcesService.saveResource(resource).subscribe((updatedResource) => {
+      this._matDialogRef.close(updatedResource);
+    });
+  }
+
+  private onFilesChangeRefOverride(files: File[]): void {
+    this.resourceInput!.isResource = true;
+    this.onFilesChangeRef?.(files);
+    this.resourceInput!.isResource = false;
+  }
+
+  private setFormValue(resource?: Resource): void {
+    if (!resource) {
+      return;
+    }
+
+    resourceConfigurationDialogFormSetValueToForm(this.formGroup, resource);
+
+    this.formGroup.controls.name.disable();
+    this.formGroup.controls.resourceType.disable();
+
+    // Reflect the changes after updating the form
+    this._changeDetectorRef.detectChanges();
+
+    // Explicitly set resourceId
+    // - resourceId is the stModel with the "resource:" prefix stripped off,
+    // - however, in our case we are using the resource name for stModel (e.g. "dummy.csv"),
+    // - so the resourceId cannot be properly evaluated internally
+    this.resourceInput!.resourceId = resource.id!;
+
+    // Invoke initResource manually
+    this.resourceInput!.initResource(resource.id!);
+
+    // Preserve the original onFilesChange
+    this.onFilesChangeRef = this.resourceInput!.onFilesChange.bind(this.resourceInput);
+
+    // Override onFilesChange
+    // - given the way the resource input is written,
+    // - it requires isResource to be false for the template
+    // - in order to render the same content as the old version,
+    // - and at the same time isResource needs to be true for
+    // - the onFilesChange handler to perform the proper action
+    this.resourceInput!.onFilesChange = this.onFilesChangeRefOverride.bind(this);
```

Comment:
In case if the code above is required, because `resource.attributes.name` doesn't contain `resource:`, it can be solved in the following way:
In component's body
```
readonly stModel$ = this.formGroup.controls.content.valueChanges
   .pipe(
      startWith(this.formGroup.controls.content.value),
      map((content) => content?.value),
      map((value) => {
           if (!value) {
                return value;
           }
           if (!value.startsWith('resource:')) {
                return `resource:${value}`;
           }
           return value;
      })
   )

```
In component's template:
```
      <step-resource-input
        #resourceInput
        label="Content"
        [stType]="formGroup.controls.resourceType.value"
        [uploadOnly]="true"
        [disableServerPath]="true"
        [stModel]="stModel$ | async"
        (stModelChange)="onContentChange($event)"
        (filesChange)="onFilesChange()"
        (uploadComplete)="onUploadComplete()"
      ></step-resource-input>

```
In that case `resourceInput` manual initialization and external method's override won't required.